### ISSUE TITLE
Investigate support for different typesets

### ISSUE THREAD

**Issue by knausj85** [knausj85/knausj_talon#174]
From https://github.com/knausj85/knausj_talon/pull/169#issuecomment-663325346:
> Since you mentioned types, one the thing I have been wanting to do with types is have the ability to toggle between default types used by the C stuff, since for instance stdint (ex: uint8_t) vs default types (unsigned char) are two common used but different typing styles across many code bases (i constantly work on different codebases). originally I tried to do this with just updating a talon setting, but it seems like you can up to a setting on the fly, so I guess we need some way of maybe standardizing non talon setting 'settings' for different programming languages. may be suitable for a different issue if you want me to file it, but I thought I'd raise it since you mentioned types here if you plan on thinking about it. I suspect this might be something ran into other languages (cpp, windows (DWORD, etc)), so may also be worth having some quasi generic method of selecting the default type set

Is changing the typeset something that could be done cleanly per-project? I have my doubts, but...

For example, the default type list could be the std_*, or whatever’s considered vanilla C. Then for projects using some other set, the list could be changed via a more specific python context.

So a python file with something like the below, plus a modified list.
```
ctx.matches = r"""
mode: user.c
mode: command 
and code.language: c
tag: ctypes_fancy
```

Then, elsewhere opt-in:

```
title: \some project\
title: \some other project\
-
tag: ctypes_fancy
```

**Comment by fidgetingbits** [comment]
it should be doable situationally so having it as an option probably makes sense, although some code bases I work in the types the original developer used will differ just between functions (likely due to copy and paste between various older projects, etc), so if I'm just adding/patching functionality here and there and not refactoring their entire code base, still having the option to just flip quickly between the type would be much more convenient, as the file itself won't often change.

**Comment by knausj85** [comment]
Makes sense. Might be a reasonable use of a mode if we need to get fancy
